common/persistence/tests/execution_mutable_state_test.go

```diff
@@ -0,0 +1,372 @@
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package tests
+
+import (
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	enumspb "go.temporal.io/api/enums/v1"
+
+	enumsspb "go.temporal.io/server/api/enums/v1"
+	persistencespb "go.temporal.io/server/api/persistence/v1"
+	"go.temporal.io/server/common/dynamicconfig"
+	"go.temporal.io/server/common/log"
+	p "go.temporal.io/server/common/persistence"
+)
+
+type (
+	executionMutableStateSuite struct {
+		suite.Suite
+		*require.Assertions
+
+		shardID     int32
+		rangeID     int64
+		namespaceID string
+		workflowID  string
+		runID       string
+
+		shardManager     p.ShardManager
+		executionManager p.ExecutionManager
+		logger           log.Logger
+	}
+)
+
+func newExecutionMutableStateSuite(
+	t *testing.T,
+	shardStore p.ShardStore,
+	executionStore p.ExecutionStore,
+	logger log.Logger,
+) *executionMutableStateSuite {
+	return &executionMutableStateSuite{
+		Assertions:   require.New(t),
+		shardManager: p.NewShardManager(shardStore),
+		executionManager: p.NewExecutionManager(
+			executionStore,
+			logger,
+			dynamicconfig.GetIntPropertyFn(4*1024*1024),
+		),
+		logger: logger,
+	}
+}
+
+func (s *executionMutableStateSuite) SetupSuite() {
+	rand.Seed(time.Now().UnixNano())
+}
+
+func (s *executionMutableStateSuite) TearDownSuite() {
+
+}
+
+func (s *executionMutableStateSuite) SetupTest() {
+	s.Assertions = require.New(s.T())
+
+	s.shardID = rand.Int31()
+	resp, err := s.shardManager.GetOrCreateShard(&p.GetOrCreateShardRequest{
+		ShardID:         s.shardID,
+		CreateIfMissing: true,
+		InitialShardInfo: &persistencespb.ShardInfo{
+			ShardId: s.shardID,
+			RangeId: 1,
+		},
+	})
+	s.NoError(err)
+	s.rangeID = resp.ShardInfo.RangeId
+
+	s.namespaceID = uuid.New().String()
+	s.workflowID = uuid.New().String()
+	s.runID = uuid.New().String()
+}
+
+func (s *executionMutableStateSuite) TearDownTest() {
+
+}
+
+func (s *executionMutableStateSuite) TestCreate_BrandNew() {
+	snapshot := randomSnapshot(
+		s.namespaceID,
+		s.workflowID,
+		s.runID,
+		rand.Int63(),
+		enumsspb.WORKFLOW_EXECUTION_STATE_CREATED,
+		enumspb.WORKFLOW_EXECUTION_STATUS_RUNNING,
+		int64(1),
+	)
+
+	_, err := s.executionManager.CreateWorkflowExecution(&p.CreateWorkflowExecutionRequest{
```

Comment:
maybe add some sanity check on the response as well?

Reply:
the response currently only contains statistics, e.g. history size
if the test also includes ^, then the test will probably explode....